runtime/native/jvm.cpp

```diff
@@ -782,6 +782,22 @@ jobjectArray getEmptyObjectArray() {
     return _EmptyObjectArray;
 }
 
+// Convert a jni type signature to its class name.
+// Rules: replace all slashes with dots.
+// e.g. [Ljava/lang/String; -> [Ljava.lang.String;
+const char* ConvertSignatureToClassName(const char* signature) {
```

Comment:
Since this is just being used for local information, we don't want to have a `malloc` in here. It would be better to take a char* as the destination argument and just supply it as a local variable in `JVM_GetClassDeclaredFields`

Otherwise your code has a memory leak

Reply:
And presumably there should be a length argument too to avoid buffer overrun